api/gcp/firestore/firestore.go

```diff
@@ -60,39 +174,39 @@ func getByDoc(ctx context.Context, doc *firestore.DocumentRef) (*FileMeta, error
 
 	snapshot, err := doc.Get(ctx)
 	if err != nil {
-		log.Println("Fail to get document from firestore:", err)
+		log.Printf("failed to get document from firestore: %v", err)
 		return nil, err
 	}
-	result, err := toResult(snapshot)
+	result, err := newFileMeta(snapshot)
 	if err != nil {
 		return nil, err
 	}
 	return result, nil
 }
 
-func toResult(snapshot *firestore.DocumentSnapshot) (*FileMeta, error) {
-	result := &FileMeta{}
-	if err := snapshot.DataTo(result); err != nil {
-		log.Println("Fail to format result snapshot:", snapshot, "err:", err)
-		return result, err
+func newFileMeta(snapshot *firestore.DocumentSnapshot) (*FileMeta, error) {
+	data := snapshot.Data()
+
+	var result = new(FileMeta)
```

Comment:
issue: remove the `new()` keyword. You can just declare `var result FileMeta` and then assign to its fields that way.